src/Stratis.Bitcoin.Features.SmartContracts/Wallet/SmartContractWalletController.cs

```diff
@@ -0,0 +1,337 @@
+                            else
+                            {
+                                // Create a record for a Call transaction
+                                Result<ContractTxData> txData = this.callDataSerializer.Deserialize(scPayment.DestinationScriptPubKey.ToBytes());
+
+                                transactionItems.Add(new ContractTransactionItem
+                                {
+                                    Amount = scPayment.Amount.ToUnit(MoneyUnit.BTC),
+                                    BlockHeight = (uint)transaction.SpendingDetails.BlockHeight,
+                                    Type = ContractTransactionItemType.ContractCall,
+                                    Hash = transaction.SpendingDetails.TransactionId,
+                                    To = txData.Value.ContractAddress.ToBase58Address(this.network)
+                                });
+                            }
+                        }
+                        else
+                        {
+                            // Create a record for every external payment sent
+                            if (transaction.SpendingDetails.Payments != null)
+                            {
+                                foreach (PaymentDetails payment in transaction.SpendingDetails.Payments)
+                                {
+                                    transactionItems.Add(new ContractTransactionItem
+                                    {
+                                        Amount = payment.Amount.ToUnit(MoneyUnit.BTC),
+                                        BlockHeight = (uint) transaction.SpendingDetails.BlockHeight,
+                                        Type = ContractTransactionItemType.Send,
+                                        Hash = transaction.SpendingDetails.TransactionId,
+                                        To = payment.DestinationAddress
+                                    });
+                                }
+                            }
+                        }
+                    }
+                }
+
+                return this.Json(transactionItems.OrderByDescending(x=>x.BlockHeight));
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, e.Message, e.ToString());
+            }
+        }
+
+        [Route("send-transaction")]
+        [HttpPost]
+        public IActionResult SendTransaction([FromBody] SendTransactionRequest request)
+        {
+            Guard.NotNull(request, nameof(request));
+
+            if (!this.ModelState.IsValid)
+                return ModelStateErrors.BuildErrorResponse(this.ModelState);
+
+            if (!this.connectionManager.ConnectedPeers.Any())
+                throw new WalletException("Can't send transaction: sending transaction requires at least one connection!");
```

Comment:
Rather than throwing an unhandled exception this should use `ErrorHelpers.BuildErrorResponse`